lib/Server.js

```diff
@@ -89,6 +89,61 @@ class Server {
       throw new Error("'filename' option must be set in lazy mode.");
     }
 
+    if (options.inline !== false) {
+      const findHMRPlugin = (config) => {
+        if (!config.plugins) {
+          return undefined;
+        }
+
+        return config.plugins.find(
+          (plugin) => plugin.constructor === webpack.HotModuleReplacementPlugin
+        );
+      };
+
+      const compilers = [];
+      const compilersWithoutHMR = [];
+      let webpackConfig;
+      if (compiler.compilers) {
+        webpackConfig = [];
+        compiler.compilers.forEach((compiler) => {
+          webpackConfig.push(compiler.options);
+          compilers.push(compiler);
+          if (!findHMRPlugin(compiler.options)) {
+            compilersWithoutHMR.push(compiler);
+          }
+        });
+      } else {
+        webpackConfig = compiler.options;
+        compilers.push(compiler);
+        if (!findHMRPlugin(compiler.options)) {
+          compilersWithoutHMR.push(compiler);
+        }
+      }
+
+      // it's possible that we should clone the config before doing
+      // this, but it seems safe not to since it actually reflects
+      // the changes we are making to the compiler
+      // important: this relies on the fact that addEntries now
+      // prevents duplicate new entries.
+      Server.addDevServerEntrypoints(webpackConfig, options);
+      compilers.forEach((compiler) => {
+        const config = compiler.options;
+        compiler.hooks.entryOption.call(config.context, config.entry);
+      });
+
+      // do not apply the plugin unless it didn't exist before.
+      if (options.hot || options.hotOnly) {
+        compilersWithoutHMR.forEach((compiler) => {
+          // addDevServerEntrypoints above should have added the plugin
+          // to the compiler options
+          const plugin = findHMRPlugin(compiler.options);
+          if (plugin) {
+            plugin.apply(compiler);
+          }
+        });
+      }
+    }
```

Comment:
I think better create helper/util for this and move this code from `Server`

Reply:
@evilebottnawi moved to `utils/updateCompiler.js`